Sources/AsyncHTTPClient/ConnectionPool/HTTP1/HTTP1ClientChannelHandler.swift

```diff
@@ -183,9 +183,19 @@ final class HTTP1ClientChannelHandler: ChannelDuplexHandler {
 
     private func run(_ action: HTTP1ConnectionStateMachine.Action, context: ChannelHandlerContext) {
         switch action {
-        case .sendRequestHead(let head, startBody: let startBody):
-            self.sendRequestHead(head, startBody: startBody, context: context)
+        case .sendRequestHead(let head, let sendEnd):
+            self.sendRequestHead(head, sendEnd: sendEnd, context: context)
+        case .notifyRequestHeadSendSuccessfully(let resumeRequestBodyStream, let startIdleTimer):
 
+            self.request!.requestHeadSent()
+            if resumeRequestBodyStream {
+                self.request!.resumeRequestBodyStream()
```

Comment:
Relatedly, why is this safe? Can it end up `nil` after the preceding outcall?

Reply:
Great catch! This is not safe. I added a test which crashed previously bit is fixed by chaining the implantation to unwrap safely: https://github.com/swift-server/async-http-client/pull/661/commits/18f6505fd957e3ac2e8aeca148faa95f05987b38